experimental/generation/generator/packages/library/src/mergeAssets.ts

```diff
@@ -294,148 +292,191 @@ async function mergeRootFile(schemaName: string, oldPath: string, oldFileList: s
+/**
+ * @description: Update generated lu file if the old lu file has been modified.
+ * @param filename Name of the lu file.
+ * @param newFileList List of new file paths.
+ * @param newPath Path to the folder of the new asset.
+ * @param mergedPath Path to the folder of the merged asset.
+ * @param delUtteranceSet Set of deleted utterance patterns.
+ * @param feedback Callback function for progress and errors.
+ */
+async function updateGeneratedLuFile(filename: string, newFileList: string[], newPath: string, mergedPath: string, delUtteranceSet: Set<string>, feedback: Feedback): Promise<void> {
+    let filePath = newFileList.filter(file => file.match(filename))[0]
+    let text = await fs.readFile(filePath, 'utf8')
+    let lines = text.split(os.EOL)
+    let resultLines: string[] = []
+    for (let line of lines) {
+        if (line.match(propertyPattern)) {
+            let newLine = await generatePatternUtterance(line)
+            if (delUtteranceSet.has(newLine)) {
+                resultLines.push(`>${line}`)
+            } else {
+                resultLines.push(line)
             }
+        } else {
+            resultLines.push(line)
+        }
+    }
+    let val = resultLines.join(os.EOL)
+    await writeToFile(newPath, mergedPath, filename, newFileList, val, feedback)
+}
 
-            const oldIntentSection = oldIntentSections[0]
-            const removedEnumValue = new Set<string>()
-            for (const enumSyn of deletedEnumValue) {
-                if (!keepEnumValue.has(enumSyn)) {
-                    removedEnumValue.add(enumSyn)
-                }
-            }
+/**
+ * @description: Generate the common pattern of the utterances.
+ * @param line Line of the lu file.
+ */
+async function generatePatternUtterance(line: string): Promise<string> {
+    let matches = line.match(propertyPattern)
+    if (matches !== null && matches !== undefined) {
+        for (let match of matches) {
+            let propertyName = match.replace('{@', '').replace('Property=', '')
+            let propertyRegExp = new RegExp(propertyName, 'g')
+            let propertyRegExpLower = new RegExp(propertyName.toLowerCase(), 'g')
+            line = line.replace(propertyRegExp, 'Dummy').replace(propertyRegExpLower, 'dummy')
+        }
+    }
+    line = line.replace('-', '').trim()
+    return line
+}
 
-            const intentBodyStatements = oldIntentSection.Body.split(os.EOL)
-            const intentResult: string[] = []
-            for (const intentBodyStatement of intentBodyStatements) {
-                let matching = false
-                for (const enumSyn of removedEnumValue) {
-                    if (intentBodyStatement.match(enumSyn)) {
-                        matching = true
-                        break
-                    }
-                }
-                if (!matching) {
-                    intentResult.push(intentBodyStatement)
+const valuePattern = /\{@([a-zA-Z0-9]+)Value=([a-zA-Z0-9\s]+)\}/g
```

Comment:
>\{@([a-zA-Z0-9]+)Value=([a-zA-Z0-9\s]+)\} [](start = 22, length = 41)

To be more robust you should use balancing expressions like this: (?<open>{@)([^=]+)=(.*)(?<-open>}) $1 will be the key and $2 the value.